internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go

```diff
@@ -127,13 +128,11 @@ func (s *Server) checkSupported(feature runtime.ModeCapability) error {
 
 func (s *Server) checkControlplane(apiName string) error {
 	switch s.Controller.Runtime().Config().Machine().Type() { //nolint:exhaustive
```

Comment:
if touching this, better to refactor using `.IsControlplane()` method of the `machine.Type`

Reply:
Dropped all irrelevant stuff.